### PR TITLE
 Log clearer handshake failures in SignalR .NET client

### PR DESCRIPTION
**Pull request by halter73**
Addresses #9327

### CODE REVIEW

File: src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs [hunk 1]
```diff
@@ -1015,14 +1015,17 @@ private async Task HandshakeAsync(ConnectionState startingConnectionState, Cance
...
                 // The other side disconnected
-                throw new InvalidOperationException("The server disconnected before the handshake was completed");
+                var ex = new IOException("The server disconnected before the handshake could be started.");
+                Log.ErrorReceivingHandshakeResponse(_logger, ex);
+                throw ex;
             }
 
             var input = startingConnectionState.Connection.Transport.Input;
 
+            using var handshakeCts = new CancellationTokenSource(HandshakeTimeout);
```
Comment (BrennanConroy) [inline src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs]: Curious, why did you move this?

File: src/SignalR/clients/csharp/Client/test/UnitTests/HubConnectionTests.ConnectionLifecycle.cs [hunk 2]
```diff
@@ -166,20 +154,43 @@ Task<ConnectionContext> ConnectionFactory(TransferFormat format)
...
+
+                using (StartVerifiableLog(ExpectedErrors))
+                {
+                    var testConnection = new TestConnection(autoHandshake: false);
+                    await AsyncUsing(CreateHubConnection(testConnection, loggerFactory: LoggerFactory), async connection =>
+                    {
+                        testConnection.Transport.Input.Complete();
+                        try
+                        {
+                            await connection.StartAsync();
```
Comment (BrennanConroy) [inline src/SignalR/clients/csharp/Client/test/UnitTests/HubConnectionTests.ConnectionLifecycle.cs]: nit: OrTimeout

File: src/SignalR/clients/csharp/Client/test/UnitTests/HubConnectionTests.Reconnect.cs [hunk 3]
```diff
@@ -17,896 +18,899 @@ namespace Microsoft.AspNetCore.SignalR.Client.Tests
 {
     public partial class HubConnectionTests
     {
-        [Fact]
-        public async Task ReconnectIsNotEnabledByDefault()
+        public class Reconnect : VerifiableLoggedTest
```
Comment (BrennanConroy) [inline src/SignalR/clients/csharp/Client/test/UnitTests/HubConnectionTests.Reconnect.cs]: Is this the only change to this file?

**Review by BrennanConroy — APPROVED** [review]

File: src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs (hunk 1 above)
Comment (halter73) [reply src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs]: I log different messages when the user manually cancels the handshake using a CancellationToken or by calling StopAsync() than I do when the handshake simply times out. I moved this out of the try scope so I could check `handshakeCts.IsCancellationRequested` in the catch block below to determine what to log.

File: src/SignalR/clients/csharp/Client/test/UnitTests/HubConnectionTests.ConnectionLifecycle.cs (hunk 2 above)
Comment (halter73) [reply src/SignalR/clients/csharp/Client/test/UnitTests/HubConnectionTests.ConnectionLifecycle.cs]: Good catch. I cleaned up most of HubConnectionTests.Protocol but missed this test despite modifying it.

File: src/SignalR/clients/csharp/Client/test/UnitTests/HubConnectionTests.Reconnect.cs (hunk 3 above)
Comment (halter73) [reply src/SignalR/clients/csharp/Client/test/UnitTests/HubConnectionTests.Reconnect.cs]: https://github.com/aspnet/AspNetCore/pull/10433/commits/5a97797b0112ba2f6b5bf45a047e0964c2cb4493#diff-071f9acf91eca9ec3e4b318ccfea3037 and removing "Reconnect" from the beginning of the test methods are the only other changes in the file. If you add "?w=1" to the URL or click the "No Whitespace" button.
